e2etest/scripts/ci.sh

```diff
@@ -7,23 +7,30 @@
 # This script is run by the docker based ci build environment and is not intended to be run separately
 # To run this for all provider tests, run ./ci-all.sh in this folder (you will need docker installed)
 
+
 SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 TESTDIR=$(realpath "${SCRIPTDIR}"/..)
-set -eouf pipefail
+# set -eouf pipefail
+set -ex
 
 # The clean up procedure is called when the script finished or is interrupted
 cleanup () {
     echo "Shutdown Parsec and clean up"
     # Stop Parsec if running
-    if [ -n "$PARSEC_PID" ]; then kill "$PARSEC_PID" || true ; fi
+    stop_service
     # Stop tpm_server if running
     if [ -n "$TPM_SRV_PID" ]; then kill "$TPM_SRV_PID" || true; fi
+    if [ -n "$TPM_MC_SRV_PID" ]; then kill "$TPM_MC_SRV_PID" || true; fi
+    # Remove the slot_number line added earlier
+    find e2etest -name "*toml" -not -name "Cargo.toml" -exec sed -i 's/^slot_number =.*/# slot_number/' {} \;
+    find e2etest -name "*toml" -not -name "Cargo.toml" -exec sed -i 's/^serial_number =.*/# serial_number/' {} \;
     # Remove fake mapping and temp files
-    rm -rf "mappings"
-    rm -f "NVChip" 
-    rm -f "${TESTDIR}/provider_cfg/tmp_config.toml"
+    rm -rf "mappings" "kim-mappings"
+    rm -f "NVChip"
+    rm -f "e2etest/provider_cfg/tmp_config.toml"
+    rm -f "parsec.sock"
 
-   echo "clean up completed"
+    if [ -z "$NO_GO_CLEAN" ]; then go clean; fi
```

Comment:
Is this variable being set somewhere? :)

Reply:
That's something that you'd set on the CLI when running the script to prevent it from cleaning up your environment when you run it, similar to [this](https://github.com/parallaxsecond/parsec/blob/main/ci.sh#L25) one for the service.